receiver/hostmetricsreceiver/config.go

```diff
@@ -24,3 +25,10 @@ type Config struct {
 	scraperhelper.ScraperControllerSettings `mapstructure:",squash"`
 	Scrapers                                map[string]internal.Config `mapstructure:"-"`
 }
+
+var _ config.CustomConfigOptions = (*Config)(nil)
+
+// Validate implements the custom validation check on configuration
```

Comment:
nit: all methods 'implement' something, we could rephrase this to something like
```
// Validate checks the receiver configuration is valid
```